frame/elections-phragmen/src/default_weights.rs

```diff
@@ -0,0 +1,87 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 2.0.0-rc6
+
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::weights::{Weight, constants::RocksDbWeight as DbWeight};
+
+impl crate::WeightInfo for () {
+	fn vote(v: u32, ) -> Weight {
+		(35890000 as Weight)
+			.saturating_add((168000 as Weight).saturating_mul(v as Weight))
+			.saturating_add(DbWeight::get().reads(5 as Weight))
+			.saturating_add(DbWeight::get().writes(2 as Weight))
+	}
+	fn vote_update(v: u32, ) -> Weight {
+		(23406000 as Weight)
+			.saturating_add((312000 as Weight).saturating_mul(v as Weight))
+			.saturating_add(DbWeight::get().reads(5 as Weight))
+			.saturating_add(DbWeight::get().writes(2 as Weight))
+	}
+	fn remove_voter() -> Weight {
+		(35814000 as Weight)
+			.saturating_add(DbWeight::get().reads(2 as Weight))
+			.saturating_add(DbWeight::get().writes(2 as Weight))
+	}
+	fn report_defunct_voter_correct(c: u32, v: u32, ) -> Weight {
+		(0 as Weight)
+			.saturating_add((1917000 as Weight).saturating_mul(c as Weight))
+			.saturating_add((28128000 as Weight).saturating_mul(v as Weight))
+			.saturating_add(DbWeight::get().reads(6 as Weight))
+			.saturating_add(DbWeight::get().writes(3 as Weight))
+	}
+	fn report_defunct_voter_incorrect(c: u32, v: u32, ) -> Weight {
+		(0 as Weight)
+			.saturating_add((1946000 as Weight).saturating_mul(c as Weight))
+			.saturating_add((28514000 as Weight).saturating_mul(v as Weight))
+			.saturating_add(DbWeight::get().reads(4 as Weight))
+	}
```

Comment:
in case of incorrect report_defunct, the reporter is removed.
the remove function does:
```rust
	fn do_remove_voter(who: &T::AccountId, unreserve: bool) {
		// remove storage and lock.
		Voting::<T>::remove(who);
		T::Currency::remove_lock(T::ModuleId::get(), who);

		if unreserve {
			T::Currency::unreserve(who, T::VotingBond::get());
		}
	}
```
so at least Voting and locks should be written no ?

Reply:
Yes this must be re-executed once more in the bench machine, I copied the weights from the other PR which has a different logic in which this operation will not remove anyone. Thanks!

Reply:
addressed.